### PR TITLE
simplify message definition.

### PR DESCRIPTION
**Pull request by drmingdrmer**
- Use a Request struct and a Reply struct for all communication.
  In a message, use req_type to identify request type such as PreAccept
  etc.

- Clarify the fields required by different message, both in comment and wiki.

proto对应的rs文件没有更新, 会乱, 等pr过了再更.

## Type of change

- **Refactoring**
- **Document changes**

# Checklist:

- [x] **Style**:       My code follows the **style guidelines** of this project
- [x] **Self-review**: I have performed a **self-review** of my own code
- [x] **Comment**:     I have **commented my code**, particularly in hard-to-understand areas
- [x] **Doc**:         I have made corresponding changes to the **documentation**
- [ ] **No-warnings**: My changes generate **no new warnings**
- [ ] **Add-test**:    I have added **tests** that prove my fix is effective or that my feature works
- [ ] **Pass**:        New and existing **unit tests pass** locally with my changes
- [ ] **Dep**:         Any **dependent** changes have been merged and published in downstream modules

### CODE REVIEW

File: wiki/replication-algo.md [hunk 1]
```diff
@@ -546,6 +546,62 @@ Non-leader replicas:
...
+- All reply messages have 3 common fields:
+  - `req_type`.
+  - `last_ballot` is the ballot number before processing the request.
+  - `instance_id`.
+
+## PreAccept request
+
+- `cmds`: the commands to run.
+- `initial_deps`: the deps when leader initiate the instance.
+- `deps_status`: a vector of committed flag of every instance in `initial_deps`
```
Comment (pengsven) [inline wiki/replication-algo.md]: preAccept request message 为什么需要  `deps_status`

Comment (drmingdrmer) [reply wiki/replication-algo.md]: 本地可能有committed的instance, 其他replica不知道, 因为fast-path提交需要all deps committed. 带过去可以减少其他instance看到自己时产生冲突的几率. 不带的话也ok能正确运行.

Comment (liubaohai) [reply wiki/replication-algo.md]: fash path需要deps都committed，是这个用处吧

Comment (pengsven) [reply wiki/replication-algo.md]: 了解，那就带吧

**Pull request merged**